packages/xo-web/src/common/xo/add-hosts-modal/index.js

```diff
@@ -41,62 +41,62 @@ import { SelectHost } from 'select-objects'
   }),
   { withRef: true }
 )
-export default class AddHostModal extends BaseComponent {
+export default class AddHostsModal extends BaseComponent {
   get value() {
-    const { nHostMissingPatches, nPoolMissingPatches } = this.state
+    const { nHostsMissingPatches, nPoolMissingPatches } = this.state
     if (
       process.env.XOA_PLAN < 2 &&
-      (nHostMissingPatches > 0 || nPoolMissingPatches > 0)
+      (nHostsMissingPatches > 0 || nPoolMissingPatches > 0)
     ) {
       return {}
     }
 
-    return { host: this.state.host }
+    return { hosts: this.state.hosts }
   }
 
   _getHostPredicate = createSelector(
     () => this.props.singleHosts,
     singleHosts => host => singleHosts[host.id]
   )
 
-  _onChangeHost = async host => {
-    if (host === null) {
+  _onChangeHosts = async hosts => {
+    if (isEmpty(hosts)) {
       this.setState({
-        host,
-        nHostMissingPatches: undefined,
+        hosts,
+        nHostsMissingPatches: undefined,
         nPoolMissingPatches: undefined,
       })
       return
     }
 
-    const { master } = this.props.pool
-    const hostMissingPatches = await getPatchesDifference(host.id, master)
-    const poolMissingPatches = await getPatchesDifference(master, host.id)
+    const { pool } = this.props
+    const hostsPatchesDiff = await getHostsPatchesDifference(pool, hosts)
+    const poolPatchesDiff = await getPoolPatchesDifference(pool, hosts)
 
     this.setState({
-      host,
-      nHostMissingPatches: hostMissingPatches.length,
-      nPoolMissingPatches: poolMissingPatches.length,
+      hosts,
+      nHostsMissingPatches: uniq(flatMap(hostsPatchesDiff, identity)).length,
```

Comment:
What's the point of not doing the `uniq/flatten` part in `getHostsPatchesDifference` directly?